### PR TITLE
fivetran-destination: add Materialize Fivetran destination

### PR DESCRIPTION
**Pull request by benesch**
This commit uses the new Fivetran SDK to build a Materialize Fivetran destination.

### Motivation

  * This PR adds a known-desirable feature.

### Tips for reviewer

~NOT READY FOR REVIEW. Will push up the full-er implementation shortly.~

### Checklist

- [x] This PR has adequate test coverage / QA involvement has been duly considered.
- [x] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [x] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [x] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [x] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  - n/a

### CODE REVIEW

**Review by ParkMyCar — APPROVED** [review]
@benesch I pushed some updates to the branch so we pass linting/CI. I reviewed the change and it looks good to me! Even though we have open questions for Fivetran I was thinking we could merge as-is and then I can followup with changes? Let me know what works for you!

**Comment by benesch** [comment]
Yeah, merging as is sounds great! You'll need to be the one to press approve as its my PR, but I endorse!

**Comment by benesch** [comment]
Oh, you did approve. Great. Marked for automerge.

**Comment by jkosh44** [comment]
It looks like I got auto-added as a reviewer, but my review is not actually needed? I'm going to remove myself, but feel free to re-add me if I was mistaken.

**Comment by benesch** [comment]
I'm so confused about what's happening in Buildkite with the submodule. Do you have any insight, Parker?

**Comment by ParkMyCar** [comment]
> I'm so confused about what's happening in Buildkite with the submodule. Do you have any insight, Parker?

I don't! That's what I am looking at now. An alternative is to copy the necessary `.proto` into our codebase, but that maybe isn't as stringent of a "link" as a submodule is.

It's especially odd that the submodule seems to get cloned in the [first run](https://buildkite.com/materialize/tests/builds/72236#018cd13d-3bc1-465d-ba1d-b021113ebab6) and then fails on the second? Hmmmm.... maybe it's some cleanup issue between runs 🤔

**Comment by benesch** [comment]
Docker image created. 🤞🏽

File: src/fivetran-destination/build.rs [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    tonic_build::configure()
+        // Enabling `emit_rerun_if_changed` will rerun the build script when
+        // anything in the include directory (..) changes. This causes quite a
+        // bit of spurious recompilation, so we disable it. The default behavior
+        // is to re-run if any file in the crate changes; that's still a bit too
+        // broad, but it's better.
+        .emit_rerun_if_changed(false)
+        .compile_with_config(
+            config,
+            &["destination_sdk.proto"],
```
Comment (aalexandrov) [inline src/fivetran-destination/build.rs:95]: @benesch this file does not appear to be part of this PR of the current `main`. How should we get it?

Comment (aalexandrov) [reply src/fivetran-destination/build.rs:95]: Self-resolving this based on an internal slack message: branches based on this PR will only work with

```
git submodule init && git submodule update --recursive 
```

**Pull request merged**